lib/spack/spack/build_systems/autotools.py

```diff
@@ -94,6 +94,9 @@ class AutotoolsPackage(PackageBase):
     #: Options to be passed to autoreconf when using the default implementation
     autoreconf_extra_args = []
 
+    #: Files to archive for packages based on autotools
+    archive_files = ['config.log']
```

Comment:
This will be wrong for some packages because config.log is typically relative to where configure runs. But we know that -AutotoolsPackage has vars for it. Should we allow substitutions?

Reply:
@adamjstewart @tgamblin Absolutely right. I have extended this behavior, and now defaults should be treated correctly even for packages that override `build_directory`